### PR TITLE
feat: add coming soon projects, list domain project with featured ones

### PR DESCRIPTION
**Pull request by enesozturk**
# Description

Adding coming soon projects and a slight UI update to the AppCard component. This PR also refactors projects fetching logic so we're now listing featured projects even though we have set an app domain.

> [!NOTE]
> CircleBadge and some of Badge component changes will be rebased once I merge the #312 

# Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/constants/projects.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { INotifyProjectWithComingSoon } from '@/utils/types'
+
+export const COMING_SOON_PROJECTS: Array<INotifyProjectWithComingSoon> = [
```
Comment (chris13524) [inline src/constants/projects.ts:3]: I don't think this info should be part of the codebase of this repo. Feels wrong to me to hardcode data in source code like this. Also we can just update that JSON file in the other repo, without needing to touch Web3Inbox?

Can we make a new GitHub repo and put a static JSON file there and `fetch()` it?

**Review by chris13524 — COMMENTED** [review]
Just a feeling I have

File: src/constants/projects.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  {
+    id: 'peanut',
+    name: 'Peanut',
+    description: 'Get notified when friends receive and claim your payments.',
+    homepage: 'https://peanut.to',
+    dapp_url: 'https://peanut.to',
+    app: {
+      browser: 'https://peanut.to'
```
[the remark this answers is not in the document]
Comment (enesozturk) [reply src/constants/projects.ts:11]: Yes, that's true, I just added them to match this array with the project types, not to create TypeScript complexity. Not a big deal for me

File: src/constants/projects.ts (hunk 1 above)
Comment (enesozturk) [reply src/constants/projects.ts:3]: This is a temporary solution (at least I hope), why maintain this basic logic in another repo? Also, it wouldn't be a big deal but there will be load time from the GH or somewhere else while fetching, which is slower than this approach.

The reason I'd chose fetching from another source would be that hardcoded solution doesn't seem quite professional.

File: src/constants/projects.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (enesozturk) [reply src/constants/projects.ts]: Updated the types and the object structure slightly, lets go this way for now soon this will be removed

**Comment by chadyj** [comment]
LGTM
🚢 

Screenshot 2024-01-05 at 10 48 30

**Review by chris13524 — APPROVED** [review]

**Pull request merged**